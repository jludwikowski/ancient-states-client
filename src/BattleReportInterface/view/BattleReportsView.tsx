import React from "react";
import BattleReport from "../../Models/BattleReport";
import {BattleReportCard} from "../components/BattleReportCard";

export class BattleReportsView extends React.Component {

    public battleReportsList:BattleReport[] = [];

    public componentDidMount(): void {
        console.log("On componentDidMount");
    };

    public onDisband(id:number) {
        console.log("On onDisband");
    };

    public render () {
        return (
            <>
                {(this.battleReportsList && this.battleReportsList.length !== 0) ?
                    this.battleReportsList.map((report, index) => (
                    <BattleReportCard report={report} key={index}/>)
                    ) : (<p>No new battle reports</p>)
                }
            </>
        )
    }

}
