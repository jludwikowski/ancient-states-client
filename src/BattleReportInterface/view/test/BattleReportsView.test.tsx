import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import { BattleReportsView } from '../BattleReportsView';

describe('BattleReportsView', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BattleReportsView />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Should match snapshot', () => {
        const component = shallow(<BattleReportsView/>);
        expect(component).toMatchSnapshot();
    });
});
