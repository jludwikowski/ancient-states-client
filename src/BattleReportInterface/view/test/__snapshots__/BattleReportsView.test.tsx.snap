// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BattleReportsView Should match snapshot 1`] = `
<Fragment>
  <p>
    No new battle reports
  </p>
</Fragment>
`;
